sky/serve/autoscalers.py

```diff
@@ -318,12 +333,42 @@ def get_latest_version_with_min_replicas(
     def select_outdated_replicas_to_scale_down(
             self, replica_infos: Iterable['replica_managers.ReplicaInfo']
     ) -> List[int]:
+        """Select outdated replicas to scale down."""
+
+        latest_ready_replicas = []
+        old_replicas = []
+        for info in replica_infos:
+            if info.version == self.latest_version:
+                if info.is_ready:
+                    latest_ready_replicas.append(info)
+            else:
+                old_replicas.append(info)
+
+        num_latest_ready_replicas = len(latest_ready_replicas)
+        if self.update_mode == serve_utils.UpdateMode.ROLLING:
+            if num_latest_ready_replicas >= self.target_num_replicas:
+                # Once the number of ready new replicas is greater than or equal
+                # to the target, we can scale down all old replicas.
+                return [info.replica_id for info in old_replicas]
+            # If rolling update is in progress, we scale down old replicas
+            # based on the number of ready new replicas.
+            num_old_replicas_to_keep = (self.target_num_replicas -
+                                        num_latest_ready_replicas)
+            old_ready_replicas = list(
+                filter(lambda info: info.is_ready, old_replicas))
+            num_old_replicas_to_scale_down = (len(old_ready_replicas) -
+                                              num_old_replicas_to_keep)
+            # We only scale down old replicas when total number of replicas
+            if num_old_replicas_to_scale_down > 0:
+                return self._select_replicas_to_scale_down(
+                    num_old_replicas_to_scale_down, old_ready_replicas)
```

Comment:
Do we only scale down from `old_ready_replicas`? what happened to old_not_ready_replicas

Reply:
We will let the old non-ready to keep provisioning, and use them if needed. Once the `num_latest_ready_replicas >= self.target_num_replicas` satisfies, we will scale down all the old replicas as in L349

Reply:
I see. So in this case we might have: 
(many old replicas still provisioning, new replicas < `self.target_num_replicas`)? Will that cause quota issue?
I am wondering if it might be fine just to keep 
total # old replicas = num_old_replicas_to_keep

Reply:
If any of the old replicas are provisioned, in the next round of `evaluate_scaling`, we will tailor the number of old replicas to the number of replicas needed based on the total ready replicas again, and it should release the quota.

I guess the same question we had for blue green update appears here as well, i.e., the old replicas that are provisioning may become READY faster than the latest ones, whether we want to have the service to fulfill all the required number of replicas faster by allowing the old replicas to keep provisioning, or we just abandon those replicas and wait a bit longer for the latest replicas to be ready.

Reply:
> we will tailor the number of old replicas to the number of replicas needed based on the total ready replicas again

My point is that "the number of old replicas" here it only applies to `old_ready_replicas`. It doesn't consider the old provisioning replicas.

Reply:
Yes, I mean once the provisioning replica goes to ready, it will be considered and tailored in the future `evaluate_scaling`.

Reply:
> Say if there are 10 provisioning old replicas, 1 old ready replica, 1 ready new replica. We need 2 ready new replicas. We might want to scale down some old provisioning replicas.

The same issue happens to our blue green update as well. In this case, we will keep the old replicas to keep provisioning, until the latest replicas have minimum number of replicas.

We can discuss whether we should change the behavior for both update mode?

Reply:
After thinking more about this, I think it makes sense to scale down the provisioning replicas, once the `target_num_replicas` is satisfied, as in that case we should just let the latest version to take over. Just updated. PTAL. @MaoZiming